drivers/hmis/lib/tasks/one_time_migration_20231121.rake

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+desc 'One time data migration from app.versions to warehouse.versions'
+# rails driver:hmis:migrate_versions_20231121
+task migrate_versions_20231121: [:environment] do
+  import_scope = GrPaperTrail::Version.where("item_type like ?", "Hmis::Hud::%")
+  OneTimeMigration20231121.new.perform(import_scope)
+end
+
+class OneTimeMigration20231121
+  def perform(import_scope)
+    # remove records from previous migration
+    GrdaWarehouse::Version.where.not(migrated_app_version_id: nil).delete_all
+
+    # GrdaWarehouse::Version.delete_all if Rails.env.development?
+    import_scope.find_in_batches.each do |batch|
+      GrdaWarehouse::Version.import!(batch.map { |r| transform_record(r) })
+    end
+  end
+
+  def transform_record(version)
+    result = version.attributes
+    # track the previous version
+    result['migrated_app_version_id'] = result.delete('id')
+    ['object', 'object_changes'].each do |field|
+      doc = result.delete(field)
+      result[field] = doc.present? ? parse_yaml(doc) : nil
+    end
+    result
+  end
+
+  def parse_yaml(doc)
+    YAML.load(
+      doc,
+      permitted_classes: [Time, Date, Symbol],
```

Comment:
@gigxz 
```suggestion
      permitted_classes: [Time, Date, Symbol, BigDecimal],
```
There will probably be others as well.

Reply:
I got successfully through my local versions with just this change. thanks

Reply:
Maybe we should just use `unsafe_load`.  This should be a pretty safe location to load from.

Reply:
Thanks, sorry I missed that! I added BigDecimal. Using `unsafe_load` is probably fine but my inclination would to be extra careful and stick with the allow-list. If we discover new types of entities that need to be included, we could always adjust the task on the fly and re-run.

Reply:
works for me.